src/app/components/room/room.service.js

```diff
@@ -144,12 +144,21 @@
             ActionService.enterRoom(msg.id, username, connection);
             // Step 3. Establish WebRTC connection with the Janus server
             // Step 4a (parallel with 4b). Publish our feed on server
+
+            var participants = 0;
             connection.publish({
+              success: function() {
+                  //start session unmuted?
+                  if(participants < 4){
```

Comment:
Yes, but what I mean is that you can also check that is not `false` in that callback.

The (rare) problem is that if the callback is really fast and gets executed before the rest of the function, `participants` won't be set at all to the correct value. I'm not sure if such a thing can happen, but...